source/src/python/PyRosetta/src/pyrosetta/bindings/pose.py

```diff
@@ -455,7 +495,7 @@ def __delitem__(self, key):
                 "Consider 'pose.scores.clear()' or 'pose.energies().clear()'" % key
             )
 
-        if not hasPoseExtraScore(self.pose, key):
+        if not hasPoseExtraScore(self.pose, key) and not hasPoseExtraScore_str(self.pose, key):
```

Comment:
This PR also fixes a bug in which setting a key with a `str`-typed value that already exists in the `Pose.scores` dictionary overwrites the existing key (as it should) but then raises a `ValueError` because the `__delitem__` method wasn't checking the `hasPoseExtraScore_str` function.